### PR TITLE
Krook collisions manufactured solutions test port

### PR DESCRIPTION
**Pull request by mrhardman**
Pull request that provides a manufactured solutions test for the charged-particle Krook collision operator in 1V and 2V. These changes also required pperp to be added to the moments and diagnostics, and upar, ppar, pperp, and vth are now tested in the post_processing.jl manufactured solutions test script.

The manufactured solutions test shows large errors when the collision frequency depends on the local density and thermal speed. This may be due to an instability. To make progress in finite time, I have added a Krook collision operator option that has a constant (in space) collision frequency. With this option, the manufactured solutions test behaves well. 

I have not supported evolving split moments options.

### CODE REVIEW

**Comment by mrhardman** [comment]
Using post_processing.jl I get an out of bounds error because of the interaction of the following two lines

https://github.com/mabarnes/moment_kinetics/blob/a8e314d439fbb221a2ff312d1762b21da62bcc14/src/coordinates.jl#L218

and 

https://github.com/mabarnes/moment_kinetics/blob/a8e314d439fbb221a2ff312d1762b21da62bcc14/src/moment_kinetics_input.jl#L371

@johnomotani What was the resolution in makie_post_processing.jl?

**Comment by johnomotani** [comment]
> Using post_processing.jl I get an out of bounds error because of the interaction of the following two lines

Ah, I didn't realise that error affected the post_processing.jl too. I have a fix somewhere, will cherry-pick onto this branch in a min.

**Comment by johnomotani** [comment]
The most recent set of test failures seem to have been because somehow the Krook operator was being turned on by default. After making sure it is off by default, the tests pass on my laptop. Fingers crossed the CI is OK again now.

**Comment by mrhardman** [comment]
> The most recent set of test failures seem to have been because somehow the Krook operator was being turned on by default. 

Thanks for correcting my mistake!

**Comment by johnomotani** [comment]
I don't think the test failures here are real. The CI servers just seem to be going super slowly today, and timing out while they are still precompiling the dependencies.

File: src/manufactured_solns.jl [hunk 1]
```diff
@@ -432,11 +517,28 @@ using IfElse
...
         if num_diss_params.z_dissipation_coefficient > 0.0 && include_num_diss_in_MMS
             Si += - num_diss_params.z_dissipation_coefficient*Dz(Dz(dfni))
         end
+        nu_krook = collisions.krook_collision_frequency_prefactor
+        if nu_krook > 0.0
+            tempi = vthi^2
+            if collisions.krook_collisions_option == "manual"
+                nuii_krook = nu_krook
+            else # default option
+                nuii_krook = nu_krook * densi * tempi^(-1.5)
```
Comment (johnomotani) [inline src/manufactured_solns.jl:531]: Change the variable name to make slightly clearer that this is not what would be 'normalised temperature' in the current conventions of the code?
```suggestion
            Ti_over_Tref = vthi^2
            if collisions.krook_collisions_option == "manual"
                nuii_krook = nu_krook
            else # default option
                nuii_krook = nu_krook * densi * Ti_over_Tref^(-1.5)
```

File: test/velocity_integral_tests.jl [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+            end
+
+            # now check that we can extract the correct moments from the distribution
+
+            dens_test = get_density(dfn,vpa,vperp)
+            upar_test = get_upar(dfn,vpa,vperp,dens_test)
+            ppar_test = get_ppar(dfn,vpa,vperp,upar_test)
+            pperp_test = get_pperp(dfn,vpa,vperp)
+            pres_test = pressure(ppar_test,pperp_test)
+            # output test results
```
Comment (johnomotani) [inline test/velocity_integral_tests.jl:84]:
```suggestion

```

File: test/velocity_integral_tests.jl [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+            for ivz in 1:vz.n
+                for ivr in 1:vr.n
+                    vz_val = vz.grid[ivz]
+                    dfn1D[ivz,ivr] = (dens/vth)*exp( - ((vz_val-upar)^2)/vth^2 )
+                end
+            end
+            dens_test = get_density(dfn1D,vz,vr)
+            upar_test = get_upar(dfn1D,vz,vr,dens_test)
+            ppar_test = get_ppar(dfn1D,vz,vr,upar_test)
+            # output test results
```
Comment (johnomotani) [inline test/velocity_integral_tests.jl:105]:
```suggestion

```

File: test/velocity_integral_tests.jl [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+                end
+            end
+
+            # now check that we can extract the correct moments from the distribution
+
+            dens_test = get_density(dfn,vpa,vperp)
+            upar_test = get_upar(dfn,vpa,vperp,dens_test)
+            ppar_test = get_ppar(dfn,vpa,vperp,upar_test)
+            pperp_test = get_pperp(dfn,vpa,vperp)
+            # output test results
```
Comment (johnomotani) [inline test/velocity_integral_tests.jl:134]:
```suggestion
```

**Review by johnomotani — APPROVED** [review]
Looks good to me. Just a couple of tiny choices left.

**Pull request merged**